presto/src/test/scala/com/yahoo/maha/executor/PrestoQueryExecutorTest.scala

```diff
@@ -0,0 +1,577 @@
+// Copyright 2017, Yahoo Holdings Inc.
+// Licensed under the terms of the Apache License 2.0. Please see LICENSE file in project root for terms.
+package com.yahoo.maha.executor.presto
+
+import java.io.{BufferedWriter, FileWriter, OutputStreamWriter}
+import java.sql.{Date, Timestamp}
+import java.util.UUID
+
+import com.yahoo.maha.core.CoreSchema._
+import com.yahoo.maha.core.FilterOperation._
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.dimension.{HivePartDimCol, _}
+import com.yahoo.maha.core.fact._
+import com.yahoo.maha.core.query._
+import com.yahoo.maha.core.query.druid.DruidQueryGenerator
+import com.yahoo.maha.core.query.presto.PrestoQueryGenerator
+import com.yahoo.maha.core.registry.RegistryBuilder
+import com.yahoo.maha.core.request._
+import com.yahoo.maha.executor.MockDruidQueryExecutor
+import com.yahoo.maha.jdbc._
+import com.yahoo.maha.report.RowCSVWriter
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers}
+
+class PrestoQueryExecutorTest extends FunSuite with Matchers with BeforeAndAfterAll with BaseQueryGeneratorTest {
+  
+  private var dataSource: Option[HikariDataSource] = None
+  private var jdbcConnection: Option[JdbcConnection] = None
+  private var prestoQueryExecutor : Option[PrestoQueryExecutor] = None
+  private val queryExecutorContext : QueryExecutorContext = new QueryExecutorContext
+  private val staticTimestamp = new Timestamp(System.currentTimeMillis())
+  private val staticTimestamp2 = new Timestamp(System.currentTimeMillis() + 1)
+
+  override protected def beforeAll(): Unit = {
+    val config = new HikariConfig()
+    config.setJdbcUrl("jdbc:h2:mem:" + UUID.randomUUID().toString.replace("-",
```

Comment:
is there any Presto Mode in h2? how can we test presto syntax here?

Reply:
Presto allows ANSI SQL, which is what default H2 mode is. There is no separate mode for Presto.